pkg/sfu/downtrack.go

```diff
@@ -362,6 +370,64 @@ func (d *DownTrack) TrackInfoAvailable() {
 	d.connectionStats.Start(ti, time.Now())
 }
 
+func (d *DownTrack) SetStreamAllocatorListener(listener DownTrackStreamAllocatorListener) {
+	d.streamAllocatorLock.Lock()
+	d.streamAllocatorListener = listener
+	d.streamAllocatorLock.Unlock()
+}
+
+func (d *DownTrack) getStreamAllocatorListener() DownTrackStreamAllocatorListener {
+	d.streamAllocatorLock.RLock()
+	defer d.streamAllocatorLock.RUnlock()
+
+	return d.streamAllocatorListener
+}
+
+func (d *DownTrack) SetStreamAllocatorReportInterval(interval time.Duration) {
+	d.ClearStreamAllocatorReportInterval()
+
+	if interval == 0 {
+		return
+	}
+
+	d.streamAllocatorLock.Lock()
+	ticker := time.NewTicker(interval)
+	d.streamAllocatorReportTicker = ticker
+	d.streamAllocatorBytesCounter.Store(0)
+
+	d.streamAllocatorReportGeneration++
+	gen := d.streamAllocatorReportGeneration
+	d.streamAllocatorLock.Unlock()
+
+	go func(generation int) {
+		<-ticker.C
```

Comment:
yep, totally missed that. Thank you for catching it. I tested that congestion recovers, guess it was just a coincidence that it recovered because of probably low rate. Thank you for catching it!!!

Reply:
Thank you @paulwe for the review and catching some embarrassing mistakes 🤦 . Have addressed your comments in this [commit](https://github.com/livekit/livekit/pull/1512/commits/48d04016a6e33d905015a34395bc2685ef1c9387). Please have a look when you get a chance.